src/ssl.c

```diff
@@ -1265,8 +1265,12 @@ int wolfSSL_send_session(WOLFSSL* ssl)
 
 /* prevent multiple mutex initializations */
 static volatile WOLFSSL_GLOBAL int initRefCount = 0;
+#ifdef WOLFSSL_PTHREADS
+static WOLFSSL_GLOBAL wolfSSL_Mutex count_mutex = PTHREAD_MUTEX_INITIALIZER;
+#else
 static WOLFSSL_GLOBAL wolfSSL_Mutex count_mutex;   /* init ref count mutex */
 static WOLFSSL_GLOBAL int count_mutex_valid = 0;
+#endif
```

Comment:
I don't like this pthread specific code. I would prefer to add a macro like `WOLFSSL_MUTEX_INIT` to the appropriate location in `wc_port.h` and `ifdef` on that to initialize the mutex. Doing it in a more generalized way, we can enable multiple platforms to be race-free by default. Since we have a threading and mutex interface, please add on to it.

Reply:
OK yeah I'll do that, but initially (heh heh) `WOLFSSL_MUTEX_INTIIALIZER` will only get defined for pthreads, since I/we don't know the initializers for all the other thread APIs.

There's two other things to consider here of course: using atomic compare-and-exchange to do the update (refactoring out the mutex entirely), and fixing any other data segment mutexes to use the initializer.  But we're making a start here at least, which will mollify Coverity.